### DISCUSSION TITLE
Recommended/optimal way for handling soft deletes?

### DISCUSSION THREAD

**Discussion by aventus13** [ravendb/ravendb#12409] [General]
Are there any recommended ways for handling soft deletes? Let's assume that all documents have the following format regardless of the collection they are in:

```
{
  "IsDeleted": false,
  // ... entity-specific properties
}
```

Is there a store/session specific setting that would allow to filter out all documents where `IsDeleted` is `true`, so that we don't need to duplicate the same condition in all queries in the application?

**Comment by ppekrol** [comment]
Hi, best way would be to use DocumentStore.OnBeforeQuery or Session.OnBeforeQuery events and modify the query there. Currently we are in a process of enhancing that scenario (https://github.com/ravendb/ravendb/pull/12391) and this should be available in nightly builds as soon as next week and in stable build in 2.5 week. Your problem is similar to the one described here https://groups.google.com/g/ravendb/c/ywgxstMdEtk/m/V9wbFmF4BAAJ I think.
